### ISSUE TITLE
[5.x] Unexpected breaking change for default field values

### ISSUE THREAD

**Issue by o1y** [statamic/cms#10002]
### Bug description

In V5 Statamic seems to always use the default value of a field, even if the field property is not present in the markdown. In V4 null was returned in this case. I'm not sure if this is an expected behavior, but this could break some templates.

### How to reproduce

1. Create a field, e.g. `page_layout` and put `{{ page_layout }}` to your Antlers template.
1. Set a default value to this field, e.g `default`.
2. Create a field condition, so the field remains hidden in the public form initially (e.g. by creating a Toggle field).
3. Save the entry. Correct Result: `page_layout` property is not saved to the markdown.

Parsing V5 (new):
* `{{ page_layout }}` parses to "default".

Parsing V4:
* `{{ page_layout }}` is null.

### Environment

```yaml
v5.0.0-beta.3
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

None

**Comment by jasonvarga** [comment]
Yes in v5 the default value you pick in blueprints will be the actual value now.

**Comment by o1y** [comment]
Ok, thanks! I’ll then migrate my templates.

**Comment by o1y** [comment]
This somehow causes more problems than it helps. 

Is it really intended behaviour to take a default value for a field which is completly hidden due to a field condition?

**Comment by duncanmcclean** [comment]
> Is it really intended behaviour to return a default value for a field which is completely hidden from the publish form?

Yes, it's intended, field conditions only apply to publish forms.

If you have conditional logic, you should probably be mirroring that in your templates, like this:

```antlers
{{ if a_toggle_that_controls_whether_another_is_shown }}
  {{ the_other_one }}
{{ /if }}
```

Instead of just this:

```antlers
{{ the_other_one }}
```
